src/http/gateway/dir-view/index.js

```diff
@@ -0,0 +1,86 @@
+'use strict'
+
+const filesize = require('filesize')
+
+const mainStyle = require('./style')
+const pathUtil = require('../utils/path')
+
+function getParentDirectoryURL (originalParts) {
+  const parts = originalParts.splice()
+
+  if (parts.length > 1) {
+    parts.pop()
+  }
+
+  return [ '', 'ipfs' ].concat(parts).join('/')
+}
+
+function buildFilesList (path, links) {
+  const rows = links.map((link) => {
+    let row = [
+      `<div class="ipfs-icon ipfs-_blank">&nbsp;</div>`,
+      `<a href="${pathUtil.joinURLParts(path, link.name)}">${link.name}</a>`,
+      filesize(link.size)
+    ]
+
+    row = row.map((cell) => `<td>${cell}</td>`).join('')
+
+    return `<tr>${row}</tr>`
+  })
+
+  return rows.join('')
+}
+
+function buildTable (path, links) {
+  const parts = pathUtil.splitPath(path)
+  const parentDirectoryURL = getParentDirectoryURL(parts)
+
+  return `
+    <table class="table table-striped">
+      <tbody>
+        <tr>
+          <td class="narrow">
+            <div class="ipfs-icon ipfs-_blank">&nbsp;</div>
+          </td>
+          <td class="padding">
+            <a href="${parentDirectoryURL}">..</a>
+          </td>
+          <td></td>
+        </tr>
+        ${buildFilesList(path, links)}
+      </tbody>
+    </table>
+  `
+}
+
+function render (path, links) {
+  return `
+    <!DOCTYPE html>
+    <html>
+    <head>
+      <meta charset="utf-8">
+      <title>${path}</title>
+      <style>${mainStyle}</style>
+    </head>
+    <body>
+      <div id="header" class="row">
+        <div class="col-xs-2">
+          <div id="logo" class="ipfs-logo"></div>
+        </div>
+      </div>
+      <br>
+      <div class="col-xs-12">
+        <div class="panel panel-default">
+          <div class="panel-heading">
+            <strong>Index of ${path}</strong>
+          </div>
+          ${buildTable(path, links)}
+        </div>
+      </div>
+    </body>
+    </html>
+  `
+}
+
+exports = module.exports
```

Comment:
it seems like a bad start to do htm templating using string manipulation. can we use an actual templating engine please?

Reply:
Seems to be only three "templates" here for the gateway, would be overkill to include a entire template engine for this.

Reply:
It feels like a bad precedent, it starts with three templates, but always grows, so I would rather have something setup properly from the start

Reply:
I agree with @dignifiedquire, worst case scenario, if complexity doesn't grow and we don't need those templates, we can easily remove them and switch back to this style, which is better than having to rework a bunch of ad-hoc templates ;)

Reply:
suggestion: http://handlebarsjs.com/

Reply:
> if complexity doesn't grow and we don't need those templates, we can easily remove them and switch back to this style

That's planning for something we don't know, rather than going with a simple solution that works now and migrate to something more complex once we really have the need for it.  Three templates (in my mind) does not justify including a template engine.

Reply:
This can be achieved as a separate PR